text/0033-view-hierarchy.md

```diff
@@ -0,0 +1,57 @@
+* Start Date: 2022-11-02
+* RFC Type: feature
+* RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+* RFC Status: -
+* RFC Driver: [Dhiogo Ramos Brustolin](https://github.com/brustolin)
+
+# Summary
+
+Allow SDKs to send a textual representation of the application's UI view hierarchy. 
+
+# Motivation
+
+Being able to see what the user was looking at is a good way to found out problems in the application. 
+This RFC proposes a way to send information from the UI state of an application, known as "view hierarchy".
+
+# Proposal 
+
+The proposal is to create a new [envelope item](https://develop.sentry.dev/sdk/envelopes/) in the form of a json representation of the UI elements. 
+This is how the envelope item header should look like:
+
+```json
+{"type":"view_hierarchy","length":...,"content_type":"application/json"}\n
+```
+
+And the envelope item payload will be a JSON array that contains a view hierarchy for each available screen/window. 
```

Comment:
How does it differentiate between multiple windows and multiple root views?

Reply:
Each Window is a UI element in the array, and since properties are dynamic, the platform may choose different properties for Windows.

Mixed UI will have different properties, but still UI Elements in a hierarchy.